src/variables.ts

```diff
@@ -0,0 +1,62 @@
+import { CustomVariableSupport, DataQueryRequest, DataQueryResponse, toDataFrame } from '@grafana/data';
+import { getTemplateSrv } from '@grafana/runtime';
+import VariableQueryEditor from './components/VariableEditor/VariableQueryEditor';
+import { DataSource, includeTimeRange } from 'DataSource';
+import { Observable, from, lastValueFrom } from 'rxjs';
+import { LogScaleQuery, LogScaleQueryType } from 'types';
+
+export class VariableSupport extends CustomVariableSupport<DataSource, LogScaleQuery> {
+  templateSrv = getTemplateSrv();
+
+  constructor(private readonly datasource: DataSource) {
+    super();
+    this.datasource = datasource;
+    this.query = this.query.bind(this);
+    this.templateSrv = getTemplateSrv();
+  }
+
+  editor = VariableQueryEditor;
+
+  query(request: DataQueryRequest<LogScaleQuery>): Observable<DataQueryResponse> {
```

Comment:
What do you think about doing this logic on the backend? calling a resource for a query seems strange to me. and this would clean up the timerange issue

Reply:
This a good point (funnily enough the inverse of a similar comment I left previously). Running this through the query method would clean up the time-range (hacky code 😅) and also provide other benefits. The only frustrating aspect is having to switch on query type in the backend but I don't _think_ that's a big deal?

Reply:
Okay I've updated this to be backend now